### ISSUE TITLE
AutolinkExtension parses emails twice

### ISSUE THREAD

**Issue by onigoetz** [thephpleague/commonmark#689]
**Version(s) affected**: 2.0.0

**Description**  
Using `CommonMarkCoreExtension` and `AutolinkExtension` together with mail addresses results in nested links

**How to reproduce**  

```php
use League\CommonMark\MarkdownConverter;
use League\CommonMark\Environment\Environment;
use League\CommonMark\Extension\Autolink\AutolinkExtension;
use League\CommonMark\Extension\CommonMark\CommonMarkCoreExtension;

$environment = new Environment([]);
$environment->addExtension(new CommonMarkCoreExtension());
$environment->addExtension(new AutolinkExtension());

$converter = new MarkdownConverter($environment);

echo $converter->convertToHtml('[me@mydomain.com](mailto:me@mydomain.com)');
```

results in

```html
<p>
  <a href="mailto:me@mydomain.com">
    <a href="mailto:me@mydomain.com">
      me@mydomain.com
    </a>
  </a>
</p>
```

But I would have expected ([babelmark too](https://babelmark.github.io/?text=%5Bme%40mydomain.com%5D(mailto%3Ame%40mydomain.com))) to get 
```html
<p>
  <a href="mailto:me@mydomain.com">
    me@mydomain.com
  </a>
</p>
```

P.S. Congrats and great work on V2 ! I'm integrating it in Daux and it's going well so far :)

**Comment by colinodell** [comment]
Good find!  It looks like this is essentially the same problem as #688, so a similar fix should work fine: #691  I'll have this merged and released sometime this week.

And thank you for the kind words!

**Comment by colinodell** [comment]
Fix has been released in 2.0.1! :tada:
